internal/services/search/search_service_resource.go

```diff
@@ -161,46 +216,219 @@ func resourceSearchServiceCreateUpdate(d *pluginsdk.ResourceData, meta interface
+			ReplicaCount:     pointer.To(replicaCount),
 		},
-		Identity: expandedIdentity,
-		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
 	}
 
-	if v, ok := d.GetOk("replica_count"); ok {
-		replicaCount := int64(v.(int))
-		properties.Properties.ReplicaCount = utils.Int64(replicaCount)
+	expandedIdentity, err := identity.ExpandSystemAssigned(d.Get("identity").([]interface{}))
+	if err != nil {
+		return fmt.Errorf("expanding `identity`: %+v", err)
 	}
 
-	if v, ok := d.GetOk("partition_count"); ok {
-		partitionCount := int64(v.(int))
-		properties.Properties.PartitionCount = utils.Int64(partitionCount)
+	// fix for issue #10151, if the identity type is TypeNone do not include it
+	// in the create call, only in the update call when 'identity' is removed from the
+	// configuration file...
+	if expandedIdentity.Type != identity.TypeNone {
+		searchService.Identity = expandedIdentity
 	}
 
-	err = client.CreateOrUpdateThenPoll(ctx, id, properties, services.CreateOrUpdateOperationOptions{})
+	err = client.CreateOrUpdateThenPoll(ctx, id, searchService, services.CreateOrUpdateOperationOptions{})
 	if err != nil {
-		return fmt.Errorf("creating/updating %s: %+v", id, err)
+		return fmt.Errorf("creating %s: %+v", id, err)
 	}
 
 	d.SetId(id.ID())
+
 	return resourceSearchServiceRead(d, meta)
 }
 
+func resourceSearchServiceUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Search.ServicesClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := services.ParseSearchServiceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, *id, services.GetOperationOptions{})
+	if err != nil {
+		return fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	if model := resp.Model; model != nil {
+		if d.HasChange("public_network_access_enabled") {
+			publicNetworkAccess := services.PublicNetworkAccessEnabled
+			if enabled := d.Get("public_network_access_enabled").(bool); !enabled {
+				publicNetworkAccess = services.PublicNetworkAccessDisabled
+			}
+
+			model.Properties.PublicNetworkAccess = pointer.To(publicNetworkAccess)
+		}
+
+		if d.HasChange("identity") {
+			expandedIdentity, err := identity.ExpandSystemAssigned(d.Get("identity").([]interface{}))
+			if err != nil {
+				return fmt.Errorf("expanding `identity`: %+v", err)
+			}
+
+			// NOTE: Passing type 'None' on update will remove all identities from the service.
+			model.Identity = expandedIdentity
+		}
+
+		if d.HasChange("hosting_mode") {
+			hostingMode := services.HostingMode(d.Get("hosting_mode").(string))
+			if pointer.From(model.Sku.Name) != services.SkuNameStandardThree && hostingMode == services.HostingModeHighDensity {
+				return fmt.Errorf("'hosting_mode' can only be set to 'highDensity' if the 'sku' is 'standard3', got %q", pointer.From(model.Sku.Name))
```

Comment:
I was told by another reviewer, in a different PR, that they wanted me to use the hard coded strings instead of the constants since it made reading the error string harder to ascertain during the review. Is there a HashiCorp coding style standard for this? This back and forth depending on the reviewer is a bit confusing as a contributor.